File: c-programming_projects/number-guessing_game.c
#include <stdio.h>
#include <string.h>
#include <time.h>
#include <stdlib.h>

#define MIN 1
#define MAX 100

int main()
{
    //Declare variables
    srand(time(0));
    int userInput;
    int guesses = 0;
    int num = (rand() % MAX) + MIN;

    do
    {
        //Prompt User for input
        printf("Enter your guess: ");
        scanf("%d", &userInput);

        //Set conditions for the guess
        if(userInput > num)
        {
            printf("Hint: %d too High, try a lower number.\n", userInput);
        }
        else if(userInput < num)
        {
            printf("Hint: %d too low, try a higher number.\n", userInput);
        }
        else
        {
            printf("%d is correct, you've won\n", userInput);
        }
        guesses++;
    
    
    } while (userInput != num);

    printf("You guessed %d times\n", guesses);
    

    printf("\nIf you found this program helpful, please leave a like.\n\nAlso, if you have any questions, contributions or corrections, please comment it.\nThank you.\n");
    


    return 0;
}

File: python-projects/number_guessing_game.py
import random

a = random.randrange(1, 10)
number = 0
guesses = 0

while not number == a:
    number = int(input("Enter a number: "))
    guesses += 1
    if number > a:
        print(f"{number} too large, try a lesser number")
    elif number < a:
        print(f"{number} too small, try a higher number")
    else:
        print("Yeah! you made it")
print(a)
print(f"You guessed {guesses} times")

print("\nIf you enjoyed this game, please leave a star/like")
print("If you have any contributions, suggestions, questions, please leave it in the comment box. \nThank you\n")
